test/e2e/specs/azure_monitor.go

```diff
@@ -0,0 +1,142 @@
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Update cluster config to disable azure monitor and update the workspace resource id")
+		*clusterConfig.Properties.MonitorProfile.Enabled = false
+		newWorkSpaceId := fmt.Sprintf("/subscriptions/225e02bc-43d0-43d1-a01a-17e584a4ef69/resourcegroups/"+
+			"%s/providers/microsoft.operationalinsights/workspaces/%s",
+			rgName, rgName)
+		*clusterConfig.Properties.MonitorProfile.WorkspaceResourceID = newWorkSpaceId
+
+		By("Executing a cluster update")
+		firstUpdate, err := azure.RPClient.OpenShiftManagedClusters.CreateOrUpdateAndWait(ctx, os.Getenv("RESOURCEGROUP"), os.Getenv("RESOURCEGROUP"), clusterConfig)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(firstUpdate).NotTo(BeNil())
+		Expect(*firstUpdate.Properties.MonitorProfile.WorkspaceResourceID).To(Equal(newWorkSpaceId))
+
+		By("Verifying that the log analytics agents are deleted")
+		err = wait.PollImmediate(2*time.Second, 5*time.Minute, func() (done bool, err error) {
+			_, err = sanity.Checker.Client.Admin.AppsV1.Deployments(azureMonitorProject).Get(clusterAgentName, metav1.GetOptions{})
+			switch {
+			case errors.IsNotFound(err):
+				return true, nil
+			case err != nil:
+				return false, err
+			}
+			return false, nil
+		})
+		Expect(err).NotTo(HaveOccurred())
+		err = wait.PollImmediate(2*time.Second, 5*time.Minute, func() (done bool, err error) {
+			_, err = sanity.Checker.Client.Admin.AppsV1.DaemonSets(azureMonitorProject).Get(nodeAgentName, metav1.GetOptions{})
+			switch {
+			case errors.IsNotFound(err):
+				return true, nil
+			case err != nil:
+				return false, err
+			}
+			return false, nil
+		})
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Update cluster config to re-enable azure monitor and the original workspace")
+		*firstUpdate.Properties.MonitorProfile.Enabled = true
+		*firstUpdate.Properties.MonitorProfile.WorkspaceResourceID = originalWorkapceId
+
+		By("Executing a cluster update")
+		secondUpdate, err := azure.RPClient.OpenShiftManagedClusters.CreateOrUpdateAndWait(ctx, os.Getenv("RESOURCEGROUP"), os.Getenv("RESOURCEGROUP"), firstUpdate)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(*secondUpdate.Properties.MonitorProfile.Enabled).To(Equal(true))
+		Expect(*secondUpdate.Properties.MonitorProfile.WorkspaceResourceID).To(Equal(originalWorkapceId))
+
+		By("Waiting for the log analytics agents to get ready")
+		err = wait.PollImmediate(2*time.Second, 5*time.Minute, ready.CheckDaemonSetIsReady(sanity.Checker.Client.Admin.AppsV1.DaemonSets(azureMonitorProject), nodeAgentName))
+		Expect(err).NotTo(HaveOccurred())
+		err = wait.PollImmediate(2*time.Second, 5*time.Minute, ready.CheckDeploymentIsReady(sanity.Checker.Client.Admin.AppsV1.Deployments(azureMonitorProject), clusterAgentName))
+		Expect(err).NotTo(HaveOccurred())
+	})
+})
+
+// withAzureMonitorIt wraps It blocks to make sure the tests are run on plugin versions supporting Azure Monitor (>= v9.0).
```

Comment:
e2e tests already implicitly require a very latest cluster - don't try to solve that problem in this PR.  Remove withAzureMonitorIt and revert the changes associated with test/manifests/config.go.